lib/state-processor/state-processor.js

```diff
@@ -9,12 +9,14 @@ const Events = require('../constants/events');
 const errorUtils = require('../errors/utils');
 
 module.exports = class StateProcessor {
-    constructor(captureProcessorType) {
+    constructor(config, captureProcessorType) {
+        this._config = config;
```

Comment:
do not store whole object, save only needed option. Such approach decreases the chance of memory leaks occurrence